Frontend_Web_Development_React_Angular_Vue/React/Week 2/React_Props_vs_State.md

```diff
@@ -0,0 +1,238 @@
+          </div>  
+      );  
+   }  
+}  
+export default App;
+
+```
+<br>
+
+#Index.js
+
+```javascript
+
+import React from 'react';  
+import ReactDOM from 'react-dom';  
+import App from './App.js';  
+  
+ReactDOM.render(
+    <App 
+        name = "World!!",
+        title = "Props"    
+     />,
+     document.getElementById('root')
+     );
+
+```
+
+    A simple component and then we passes the props as attributes and then access them inside our component using this.props. So props makes components reusable by giving components the ability to receive data from the parent component in the form of props. They are immutable.
+
+### **Output :-**
+<br>
+
+<img src="assets/../../../assets/Output1.png" width="400px">
+
+<hr>
+
+## Props in Functional Component :-
+<br>
+
+### **Example :-**
+<br>
+
+#App.js
+
+```javascript
+
+function App(props) {
+  return (
+    <div>  
+        <h1> Hello { this.props.name } </h1> 
+        <p> 
+            { props.title } in Functional Component.
+        </p>            
+    </div>
+  );
+}
+
+```
+<br>
+
+#Index.js
+
+```javascript
+
+import React from 'react';  
+import ReactDOM from 'react-dom';  
+import App from './App.js';  
+  
+ReactDOM.render(
+    <App 
+        name = "World!!",
+        title = "Props"    
+     />,
+     document.getElementById('root')
+     );
+
+```
+
+    we passes the props as attributes and then access them inside our component using props.title, So props makes components reusable by giving components the ability to receive data from the parent component in the form of props. They are immutable.
+
+### **Output :-**
+<br>
+
+<img src="assets/../../../assets/Output_2.png" width="400px">
+
+<hr>
+<br>
+
+## React State :-
+<br>
+
+* React components has a built-in state object.
+
+* The state object is where you store property values that belongs to the component.
+
+* When the state object changes, the component re-renders.
+
+## Creating the state Object :- 
+<br>
+
+```javascript
+
+class Car extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {title: "Morning"};
+  }
+  render() {
+    return (
+      <div>
+        <h1>Good Morning!</h1>
+      </div>
+    );
+  }
+}
+
+```
+<br>
+
+## Using the state Object :-
+<br>
+
+```javascript
+
+class Car extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {title: "Morning"};
+  }
+  render() {
+    return (
+      <div>
+        <h1>Good {this.state.title}!</h1>
+      </div>
+    );
+  }
+}
+
+```
+<br>
+
+## Changing the state Object :-
+<br>
+
+```javascript
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {title: "Morning"};
+  }
+    changeTitle = () => {
+    this.setState({Title : "Night"});
+
+  render() {
+    return (
+      <div>
+        <h1>Good {this.state.title}!</h1>
+      </div>
+      <button type="button" onClick={this.changeTitle}>Change Title</button>
```

Comment:
Wrap up the jsx components into a single element.

Reply:
@mitraditya okk sir